tempto-core/src/main/java/com/teradata/tempto/fulfillment/table/TableDefinitionsRepository.java

```diff
@@ -99,14 +101,73 @@ public TableDefinitionsRepository(Collection<TableDefinition> tableDefinitions)
 
     public <T extends TableDefinition> T register(T tableDefinition)
     {
-        checkState(!tableDefinitions.containsKey(tableDefinition.getName()), "duplicated table definition: %s", tableDefinition.getName());
-        tableDefinitions.put(tableDefinition.getName(), tableDefinition);
+        TableDefinitionRepositoryKey repositoryKey = asRepositoryKey(tableDefinition.getTableHandle());
+        checkState(!tableDefinitions.containsKey(repositoryKey), "duplicated table definition: %s", repositoryKey);
+        tableDefinitions.put(repositoryKey, tableDefinition);
         return tableDefinition;
     }
 
-    public TableDefinition get(String name)
+    public TableDefinition get(TableHandle tableHandle)
```

Comment:
Why not store table in repository with schema/catalog name or an alias?
E.g. TableDefinition(name = "nation", uniqueName = "tpch.nation").
Then you refer to that table by `uniqueName` but table name in DB would be based on `name`. When there's a clash (both tpch.nation and tpcds.nation are used in the same schema - there's a runtime error during test execution, so you have to rewrite your test to not use tables of the same name in single test in the same schema).

What you proposed is full of implicit behavior which can be very misleading. I'd say we should always be as much explicit as possible in testing.